lib/util/http/index.ts

```diff
@@ -0,0 +1,118 @@
+import is from '@sindresorhus/is/dist';
+import URL from 'url';
+import got from '../got';
+
+interface OutgoingHttpHeaders {
+  [header: string]: number | string | string[] | undefined;
+}
+
+export interface HttpOptions {
+  auth?: string;
+  baseUrl?: string;
+  headers?: OutgoingHttpHeaders;
```

Comment:
Do we want to restrict which headers we allow, so that they're all known? Or would that be too much typing?

Reply:
depends on how many headers we are currently using 🤔

Reply:
Maybe let's skip for now